modules/transformer-batch/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/transformer/batch/ShredJob.scala

```diff
@@ -325,9 +325,19 @@ object ShredJob {
 
           Transformer.WideRowParquetTransformer(allFields, schema)
       }
+      val beginning = System.currentTimeMillis()
       val job = new ShredJob(spark, transformer, config, resolverConfig)
       val completed = job.run(folder.folderName, eventsManifest)
       Discovery.seal(completed, sendToQueue, putToS3, config.featureFlags.legacyMessageFormat)
+
+      config.monitoring.metrics.cloudwatch.foreach { metrics =>
+        Metrics.sendDuration(
+          metrics.namespace,
+          metrics.transformDuration,
+          (System.currentTimeMillis() - beginning).millis,
```

Comment:
any chance of context switching at thread level leading to a higher transformDuration here?

Reply:
Well, we're not after the millisecond here. And it doesn't matter much were we take the metrics, as long as this is always from the same place.

Reply:
fair enough, thanks!